### ISSUE TITLE
Session closed when body size exceeds server limit

### ISSUE THREAD

**Issue by SzigetiPeter** [strophe/strophejs#249]
When the body size exceeds _max_stanza_size_ set on server then the session is terminated. I want to fix this issue client side by sending multiple smaller bodies in _onIdle: callback function. 
I have considered two approaches:
1. If the body exceeds a settable size limit then the _onIdle _for_ loop constructing the body will break. This implies that the next body will be sent only on the next idle cycle. 
2. Send multiple Strophe.Requests from one _onIdle cycle based on size separation as in previous point. 

What are your recommendations and what pull request would you accept?

**Comment by jcbrand** [comment]
From what I can tell, adding new Strophe.Request objects to the `_requests` queue is a relatively low-level operation that only happens in a few cases (connecting, terminating, no-auth-received and when idle).

Stanzas are added to the `connection._data` queue and then converted into a single Strophe.Request object in `_onIdle`.

It's not clear to me what would now happen if a bunch of extra requests are pushed into the queue, it seems safer to me to keep them in the `connection._data` queue for processing during the next idle cycle.

So to summarize, I think the 1st option is preferable, in other words, to keep the stanzas that are not included in the current body in the `connection._data` queue for processing during the next idle cycle.
